test/types/list_collections.test-d.ts

```diff
@@ -8,38 +8,30 @@ const db = new MongoClient('').db();
 type EitherCollectionInfoResult = CollectionInfo | Pick<CollectionInfo, 'name' | 'type'>;
 
 // We default to the CollectionInfo result type
-expectType<ListCollectionsCursor<Pick<CollectionInfo, 'name' | 'type'> | CollectionInfo>>(
-  db.listCollections()
-);
+expectType<ListCollectionsCursor<CollectionInfo>>(db.listCollections());
 // By default it isn't narrowed to either type
 expectNotType<ListCollectionsCursor<Pick<CollectionInfo, 'name' | 'type'>>>(db.listCollections());
-expectNotType<ListCollectionsCursor<CollectionInfo>>(db.listCollections());
+expectType<ListCollectionsCursor<CollectionInfo>>(db.listCollections());
 
 // Testing each argument variation
 db.listCollections();
 db.listCollections({ a: 2 });
 db.listCollections({ a: 2 }, { batchSize: 2 });
 
 const collections = await db.listCollections().toArray();
-expectType<EitherCollectionInfoResult[]>(collections);
+expectNotType<EitherCollectionInfoResult[]>(collections);
```

Comment:
I think adding an expectType ahead of this line would help make the intention more clear